sdk/batch/Azure.Compute.Batch/tests/Infrastructure/BatchLiveTestBase.cs

```diff
@@ -0,0 +1,117 @@
+            UseDefaultGuidFormatForClientRequestId = true;
+        }
+
+        public BatchLiveTestBase(bool isAsync) : base(isAsync)
+        {
+            SanitizedHeaders.Add("client-request-id");
+            UseDefaultGuidFormatForClientRequestId = true;
+        }
+
+        public bool isPlayBack()
+        {
+            return this.Mode == RecordedTestMode.Playback;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BatchClient" /> with the endpoint and API key provided via environment
+        /// variables and instruments it to make use of the Azure Core Test Framework functionalities.
+        /// </summary>
+        /// <param name="useTokenCredential">Whether or not to use a <see cref="TokenCredential"/> to authenticate. An <see cref="AzureKeyCredential"/> is used by default.</param>
+        /// <param name="skipInstrumenting">Whether or not instrumenting should be skipped. Avoid skipping it as much as possible.</param>
+        /// <returns>The instrumented <see cref="BatchClient" />.</returns>
+        public BatchClient CreateBatchClient(TestAuthMethods testAuthMethod = TestAuthMethods.ClientSecret, bool skipInstrumenting = false)
+        {
+            var options = InstrumentClientOptions(new BatchClientOptions());
+            BatchClient client;
+            Uri uri = new Uri("https://" + TestEnvironment.BatchAccountURI);
+
+            var authorityHost = TestEnvironment.AuthorityHostUrl;
+            switch (testAuthMethod)
+            {
+                case TestAuthMethods.ClientSecret:
+                    {
+                        client = new BatchClient(uri, TestEnvironment.Credential, options);
+                    }
+                    break;
+                case TestAuthMethods.NamedKey:
+                    {
+                        var credential = new AzureNamedKeyCredential(TestEnvironment.BatchAccountName, TestEnvironment.BatchAccountKey);
+                        client = new BatchClient(uri, credential, options);
+                    }
+                    break;
+                default:
+                    {
+                        var credential = new DefaultAzureCredential();
+                        client = new BatchClient(uri, credential, options);
+                    }
+                    break;
+            }
+            return skipInstrumenting ? client : InstrumentClient(client);
+        }
+
+        /// <summary>
+        /// Poll all the tasks in the given job and wait for them to reach the completed state.
+        /// </summary>
+        /// <param name="jobId">The ID of the job to poll</param>
+        /// <returns>A task that will complete when all Batch tasks have completed.</returns>
+        /// <exception cref="TimeoutException">Thrown if all tasks haven't reached the completed state after a certain period of time</exception>
+        public async Task waitForTasksToComplete(BatchClient client, String jobId, bool isPlayBackMode = false)
```

Comment:
Why are we not following .NET standards for naming?   This should be PascalCase.

Reply:
fixed